hasher-matcher-actioner/hmalib/common/mappings.py

```diff
@@ -1,37 +1,206 @@
-we must provide a mapping in hmalib land.
+The get_pytx_functionality_mapping() will be called from almost all entrypoints.
+Entrypoints usually, are the modules where a lambda_handler is defined.
 
-If you find yourself needing a map between hmalib types or hmalib and
-python-threatexchange types, they should go here.
-
-If you find yourself needing a map between python-threatexchange types alone,
-they should go into python-threatexchange. eg. threatexchange.content_type.meta. 
+Everything that is configurable from threatexchange.meta is stored in dynamodb.
+This module also exposes mechanisms to create/edit/delete those configurables. 
 """
+
+from dataclasses import dataclass, field
+import importlib
 import typing as t
+from numpy import append
 
-from threatexchange.signal_type.pdq import PdqSignal
-from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.meta import (
+    FunctionalityMapping,
+    SignalTypeMapping,
+    FetcherMapping,
+    CollaborationConfigStoreBase,
+)
 from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.fetcher.fetch_api import SignalExchangeAPI
+from threatexchange.fetcher.apis.fb_threatexchange_api import (
+    FBThreatExchangeSignalExchangeAPI,
+)
+from threatexchange.content_type.photo import PhotoContent
+from threatexchange.content_type.video import VideoContent
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.signal_type.pdq import PdqSignal
+
+from hmalib.common.config import HMAConfig, create_config
+from hmalib.common.logging import get_logger
+from hmalib.aws_secrets import AWSSecrets
 
-from hmalib.indexers.s3_indexers import (
-    S3BackedMD5Index,
-    S3BackedPDQIndex,
-    S3BackedPDQFlatIndex,
-    S3BackedInstrumentedIndexMixin,
+# Used if this is the first time HMA is being booted. TODO: Should this be
+# configurable before boot?
+DEFAULT_SIGNAL_AND_CONTENT_TYPES = SignalTypeMapping(
+    content_types=[PhotoContent, VideoContent], signal_types=[VideoMD5Signal, PdqSignal]
 )
 
-# Maps from signal type → [index] i.e. list of indexes the can be use for that signal type.
-INDEX_MAPPING: t.Dict[
-    t.Type[SignalType], t.List[t.Type[S3BackedInstrumentedIndexMixin]]
-] = {
-    PdqSignal: [S3BackedPDQIndex, S3BackedPDQFlatIndex],
-    VideoMD5Signal: [S3BackedMD5Index],
-}
+logger = get_logger(__name__)
+
+
+def import_class(full_class_name) -> t.Type:
+    module_name, klass = full_class_name.rsplit(".", 1)
+    module = importlib.import_module(module_name)
+    return getattr(module, klass)
+
+
+@dataclass
+class ToggleableSignalTypeConfig(HMAConfig):
+    signal_type_class: str
+
+    # Allow soft disables.
+    enabled: bool = True
+
+
+@dataclass
+class ToggleableContentTypeConfig(HMAConfig):
+    content_type_class: str
+
+    # Allow soft disables.
+    enabled: bool = True
+
+
+class HMASignalTypeMapping(SignalTypeMapping):
```

Comment:
It is news to me that no interface will take SignalTypeMapping as an argument. I thought surely FunctionalityMapping and children would be the weapon of choice when encapsulating functionality. I don't mind it either way.

To reduce unintended dependencies, I'll take up your suggestion of making a 'contains' check rather than an 'is a` check.